primitives/src/circuit/tree/structs.rs

```diff
@@ -86,6 +86,41 @@ impl MembershipProofVar {
         Ok(Self::new(path_elements))
     }
 
+    /// Create a new membership proof variable from a vector of variables.
+    pub fn from_vars<F: PrimeField>(
+        circuit: &mut PlonkCircuit<F>,
+        vars: &[Variable],
+    ) -> Result<Self, CircuitError> {
+        if vars.is_empty() {
```

Comment:
Do we accept the empty path? With vars.len() == 2:

vars.is_empty() → false

vars.len() % 2 != 0 → false (so it passes the “even length” check)

path_vars = vars[1..vars.len()-1] = vars[1..1] → empty slice

chunks_exact(2) over an empty slice yields zero chunks

return Ok(MembershipProofVar { path_elements: vec![] })